buildAndReleaseTask/pulumi.ts

```diff
@@ -8,13 +8,58 @@ import { PULUMI_ACCESS_TOKEN } from "./vars";
 
 interface IEnvMap { [key: string]: string; }
 
-async function selectStack(toolPath: string, pulExecOptions: tr.IExecOptions) {
+interface IExecResult {
+    exitCode: number;
+    execErr?: string;
+}
+
+async function runSelectStack(
+    pulStack: string,
+    toolPath: string,
+    pulExecOptions: tr.IExecOptions): Promise<IExecResult> {
+    const stackSelectRunner = tl.tool(toolPath);
+    let execErr = "";
+    // The toolrunner emits an event that contains the error message line from the tool
+    // that runs a command. Listen for that event and update the execution error message
+    // accordingly.
+    stackSelectRunner.on("errline", (errLine: string) => execErr = errLine);
+
+    const exitCode = await stackSelectRunner.arg(["stack", "select", pulStack!]).exec(pulExecOptions);
+    return { exitCode, execErr };
+}
+
+/**
+ * Selects the stack specified as input to this task. If the selection command
+ * fails, then creates the stack if the `createStack` task input is `true`,
+ * IFF the stack selection failure was due to the stack not being found.
+ */
+async function selectOrCreateStack(toolPath: string, pulExecOptions: tr.IExecOptions) {
     const pulStack = tl.getInput("stack", true);
-    const exitCode = await tl.exec(toolPath, ["stack", "select", pulStack], pulExecOptions);
-    if (exitCode !== 0) {
+    const selectStackExecResult = await runSelectStack(pulStack!, toolPath, pulExecOptions);
+    if (selectStackExecResult.exitCode === 0) {
+        return;
+    }
+
+    // Check if the user requested to create the stack if it does not exist.
+    const createStack = tl.getBoolInput("createStack");
+    if (!createStack) {
         tl.setResult(tl.TaskResult.Failed, tl.loc("PulumiStackSelectFailed", pulStack));
         return;
     }
+
+    // Check if the error was because the stack was not found.
+    const errMsg = `no stack named '${pulStack}' found`;
```

Comment:
Unfortunately, I don't have a more elegant way to do this, due to the lack of a query option in the CLI. This does make this a bit brittle. However, should the error message be changed and this stops working, the side effect is that users who specify an unknown stack name would just start seeing failures, rather than something else, that is unexpected.

Reply:
So https://github.com/pulumi/pulumi/pull/3852 was merged 2 days ago. Once it is released, using the built-in way for newer CLI versions and falling back to this method would be the way to go.

Reply:
Note: https://github.com/pulumi/pulumi/pull/3852 is now available in v1.10.0, which was released yesterday. (Though, due to a recent regression in our AppVeyor config, the release was not published for Windows. I think we're going to release v1.10.1 today to get the Windows release out.)